src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,160 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.hash = (key == null) ? 0 : key.hashCode();
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || table.length == 0) {
+            table = resize();
+        } else if (size >= threshold) {
+            Node<K, V> [] oldTable = table;
+            table = resize();
+            table = transferBuckets(oldTable);
+        }
+        int bucket = hash(key);
+        if (table[bucket] == null) {
+            table[bucket] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> node = table[bucket];
+            while (node != null) {
+                if (node.key == null && key == null) {
+                    node.value = value;
+                    break;
+                }
+                if (node.key != null && key != null) {
+                    if ((node.key.hashCode() == key.hashCode()) && (node.key.equals(key))) {
+                        node.value = value;
+                        break;
+                    }
+                }
+                if (node.next == null) {
+                    node.next = new Node<>(key, value, null);
+                    break;
+                } else {
+                    node = node.next;
+                }
+            }
+        }
+        size = getSize();
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
```

Comment:
add a consturctor for hashmap, initialize table there, remove this check in methods